### PR TITLE
Handle backwards compatibility

### PR DESCRIPTION
**Pull request by hcarneiro**
- This brings the settings from old themes to the new version

### CODE REVIEW

File: src/Application.vue [hunk 1]
```diff
@@ -157,6 +158,12 @@ export default {
         setThemeInstance(theme.instances[0])
         setActiveTheme(theme)
 
+        // There are old settings apply them to the new theme
```
Comment (squallstar) [inline src/Application.vue]: how are you handling failures if the network request fails?

File: src/Application.vue [hunk 2]
```diff
@@ -196,7 +203,24 @@ export default {
       // Automatically create a theme instance if one doesn't exist
       if (themeWithoutInstances == themes.length) {
         const flipletTheme = _.find(themes, { name: FLIPLET_THEME })
-        this.createDefaultInstance(flipletTheme.id, toReuse)
+
+        // Checks for older versions
+        ThemeModel.getAllVersions()
+          .then((result) => {
+            const allThemes = result.widgets
+            const versionOneTheme = _.find(allThemes, { name: 'Bootstrap' })
```
Comment (squallstar) [inline src/Application.vue]: surely you also want to make sure the version is 1.0.0?

File: src/resources/theme.js [hunk 3]
```diff
@@ -11,6 +11,16 @@ export default {
       }
     })
   },
+  getAllVersions() {
+    return Fliplet.Env.get('development') ? Promise.resolve() : Fliplet.API.request({
+      url: [
```
Comment (squallstar) [inline src/resources/theme.js]: why don't you search by package? I think we're only looking for one specific theme, correct?

Comment (squallstar) [reply src/resources/theme.js]: `package=com.fliplet.theme.default`

File: src/Application.vue (hunk 1 above)
Comment (hcarneiro) [reply src/Application.vue]: @squallstar There is a `catch()` where I do the request and the error is presented in the UI.
Also if that fails everything else stops, so therefore the user will need to try again.

**Pull request merged**